packages/camera/camera/lib/src/camera_controller.dart

```diff
@@ -172,6 +172,8 @@ class CameraValue {
     CameraDescription? description,
     Optional<DeviceOrientation>? previewPauseOrientation,
   }) {
+    assert((isRecordingVideo ?? false) ==
```

Comment:
Is this assert in `copyWith` necessary? This is used by other methods to freely change state, so I don't think we should be asserting anything here. Potentially this check is better in the method that changes it.

Reply:
@stuartmorgan [needs this invariant to hold](https://github.com/flutter/packages/commit/8e8091365ea1e3a4eeb141a39cc35a8f4e71a8ee#diff-811158e46b3c8346bd837b73d34dd132a06f8246822b254420ec7df586550789R80), so I thought rather do an assert when at the time invariant is broken, rather than have `CameraValues` floating around that will cause crashes some time in the future. That would certainly helped me at least.